upload_certificate.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+import json
+import sys
+import base64
+from typing import Mapping, Sequence, Any
+
+import requests
+
+
+# build headers
+def build_cloudflare_request_headers(auth_email: str, auth_key: str) -> Mapping[str, str]:
+    return {
+        "Content-Type": "application/json",
+        "X-Auth-Key": auth_key,
+        "X-Auth-Email": auth_email
+    }
+
+
+# upload certificate to cloudflare
+def upload_certificate(zone_id: str, auth_email: str, auth_key: str, key: str, crt: str):
+    url: str = f"{CF_BASE_URL}/zones/{zone_id}"
+    certificates_url: str = url + '/custom_certificates'
+
+    certificate: dict = {
+        'certificate': crt,
+        'private_key': key
+    }
+
+    requests.post(url=certificates_url,
+                  headers=build_cloudflare_request_headers(auth_email=auth_email, auth_key=auth_key),
+                  json=certificate).raise_for_status()
```

Comment:
Per your suggestion, lets also check if the certificate exists; if so, compare it and update if necessary. Otherwise, create (as you do now).